providers/google/src/airflow/providers/google/cloud/hooks/cloud_composer.py

```diff
@@ -472,6 +473,32 @@ def trigger_dag_run(
 
         return response.json()
 
+    def pull_dag_runs(
+        self,
+        composer_airflow_uri: str,
+        composer_dag_id: str,
+        timeout: float | None = None,
+    ) -> dict:
+        """
+        Pull the list of dag runs for provided DAG.
+
+        :param composer_airflow_uri: The URI of the Apache Airflow Web UI hosted within Composer environment.
+        :param composer_dag_id: The ID of DAG.
+        :param timeout: The timeout for this request.
+        """
+        response = self.make_composer_airflow_api_request(
+            method="GET",
+            airflow_uri=composer_airflow_uri,
+            path=f"/api/v1/dags/{composer_dag_id}/dagRuns",
+            timeout=timeout,
+        )
+
+        if response.status_code != 200:
+            self.log.error(response.text)
```

Comment:
```suggestion
            self.log.error("Failed to pull DAG runs for dag_id=%s from %s (status=%s): %s", composer_dag_id, composer_airflow_uri, response.status_code, response.text)
```

nit: a clearer log message here (including DAG ID, URI, and status) would make failures easier to debug.

Reply:
@Programmer-RD-AI good point, I have updated the message